### PR TITLE
Add some changes for supporting MSFT kernel

### PR DESCRIPTION
**Pull request by lubaihua33**
Test results - 
```
[LISAv2 Test Results Summary]
Test Run On           : 03/25/2020 07:12:39​
ARM Image Under Test  : Canonical : UbuntuServer : 18.04-LTS : latest​
Initial Kernel Version: 5.0.0-1032-azure​
Final Kernel Version  : 4.19.84-ea0938a64e79​
Total Test Cases      : 1 (1 Passed, 0 Failed, 0 Aborted, 0 Skipped)​
Total Time (dd:hh:mm) : 0:0:16​
​
   ID TestArea             TestCaseName                                                                TestResult TestDuration(in minutes) ​
-------------------------------------------------------------------------------------------------------------------------------------------​
    1 LKS                  LINUX-KERNEL-SELFTESTS                                                            PASS                 10.7 ​
total test cases : 86 ​
total passed : 70 ​
total failed : 0 ​
total skipped : 16 
```

### CODE REVIEW

**Comment by LiliDeng** [comment]
Do you need run any regression testing against other distro?

**Review by juhlee-microsoft — APPROVED** [review]

**Comment by lubaihua33** [comment]
···
[LISAv2 Test Results Summary]
Test Run On           : 03/26/2020 02:50:01​
ARM Image Under Test  : OpenLogic : CentOS : 8.0 : latest​
Initial Kernel Version: 4.19.90+​
Final Kernel Version  : 4.19.90+​
Total Test Cases      : 1 (0 Passed, 1 Failed, 0 Aborted, 0 Skipped)​
Total Time (dd:hh:mm) : 0:0:10​
​
   ID TestArea             TestCaseName                                                                TestResult TestDuration(in minutes) ​
-------------------------------------------------------------------------------------------------------------------------------------------​
    1 LKS                  LINUX-KERNEL-SELFTESTS                                                            FAIL                10.24 ​
ipc_msgque : FAIL ​
proc_proc-self-map-files-002 : FAIL ​
total test cases : 82 ​
total passed : 67 ​
total failed : 2 ​
total skipped : 13 
···

For the failed test case, one of them I have filed a bug, the other is the kernel self tests' own issue.

File: Testscripts/Windows/Run-LksSuite.ps1 [hunk 1]
```diff
@@ -36,21 +36,20 @@ function Get-TestResult ($currentTestResult, $currentTestData) {
                 $metaData = $_.split(":")[1].replace(' ','').split('[')[0]
                 $result = $_.split("[]")[1]
                 $item = "{0,-35} {1,10}" -f $metaData, $result
-            } elseif ($_ -match "(ok|not ok)+\s+\d+\s+[\w]") {
-                $metaData = $_.split(" ")[2]
-                $result = $_.split(" ")[0]
-
-                if ($result -eq "ok") {
+            } elseif ($_ -match "(ok|not ok)+\s+\d+\s+(selftests:)\s+[\w|\w-\w:]+\s+[\w|\w-\w|.]") {
+                if ($_.Contains("not ok")) {
```
Comment (LiliDeng) [inline Testscripts/Windows/Run-LksSuite.ps1:40]: why not  ok == skip?

Comment (lubaihua33) [reply Testscripts/Windows/Run-LksSuite.ps1:40]: The results of 5.4 kernel selftests have this format report:
```not ok 1 selftests: binderfs: binderfs_test # SKIP```
So we check if the line contains "SKIP", if so we flag this test case result as "SKIP".

File: Testscripts/Linux/Linux-Kernel-Self-Tests.sh [hunk 2]
```diff
@@ -258,11 +258,11 @@ function build_and_run_lks() {
...
         TOTAL_TARGETS="$targets"
+        if [[ "$TARGETS" != "" ]]; then
+            check_targets "$targets" "$TARGETS"
+        fi
 
-        if [[ "$CUSTOM_KERNEL_FLAG" != "TRUE" && "$DISTRO_KERNEL_FLAG" != "TRUE" ]]; then
-            if [[ "$TARGETS" != "" ]]; then 
-                check_targets "$targets" "$TARGETS"
-            fi
+	if [[ "$CUSTOM_KERNEL_FLAG" != "TRUE" && "$DISTRO_KERNEL_FLAG" != "TRUE" ]]; then
```
Comment (LiliDeng) [inline Testscripts/Linux/Linux-Kernel-Self-Tests.sh]: indent?

Comment (lubaihua33) [reply Testscripts/Linux/Linux-Kernel-Self-Tests.sh]: I have a new commit to modify this.

**Review by LiliDeng — APPROVED** [review]

**Pull request merged**